code/modules/projectiles/projectile/special.dm

```diff
@@ -141,7 +151,11 @@ obj/item/projectile/kinetic/New()
 		var/turf/simulated/mineral/M = target_turf
 		M.gets_drilled(firer)
 	new /obj/item/effect/kinetic_blast(target_turf)
-
+	if(src.splash >= 1)

```

Comment:
You know, since you would never have a negative value for "splash", you could just use `if(spash)` for the same result.